ThunderRequest/RequestController.swift

```diff
@@ -0,0 +1,736 @@
+        ephemeralSession.invalidateAndCancel()
+        resetSessions()
+    }
+    
+    /// Cancels requests with a specific tag in any of the request queues,
+    /// calling the completion with a cancellation error
+    ///
+    /// - Parameter tag: The tag to cancel the requests for
+    public func cancelRequestsWith(tag: Int) {
+        
+        defaultSession.getAllTasks { (tasks) in
+            tasks.filter({ $0.tag == tag }).forEach({ (task) in
+                task.cancel()
+            })
+        }
+        
+        backgroundSession.getAllTasks { (tasks) in
+            tasks.filter({ $0.tag == tag }).forEach({ (task) in
+                task.cancel()
+            })
+        }
+        
+        ephemeralSession.getAllTasks { (tasks) in
+            tasks.filter({ $0.tag == tag }).forEach({ (task) in
+                task.cancel()
+            })
+        }
+    }
+    
+    /// Calls invalidateAndCancel on all internal `URLSession` objects
+    /// to allow self to be deallocated
+    public func invalidateAndCancel() {
+        defaultSession.invalidateAndCancel()
+        backgroundSession.invalidateAndCancel()
+        ephemeralSession.invalidateAndCancel()
+    }
+    
+    /// Sets the shared request credentials, optionally saving them to the keychain.
+    ///
+    /// - Parameters:
+    ///   - sharedRequestCredentials: The request credential to set/save.
+    ///   - savingToKeychain: Whether or not to save the credentials to the keychain.
+    ///   - accessibility: The accessibility of the credential in the keychain.
+    ///
+    /// - Note: If a `OAuth2Credential` object is stored to the keychain by this method
+    /// it will be fetched from the keychain each time an `OAuth2Delegate` with the same
+    /// service identifier is set on the request controller. If `OAuth2Delegate` is non-nil
+    /// when this method is called it will be saved under the current delegate's service
+    /// identifier. Otherwise it will be saved under a string appended by `sharedBaseURL`
+    public func set(sharedRequestCredentials: RequestCredential?, savingToKeychain: Bool, accessibility: CredentialStore.Accessibility = .afterFirstUnlock) {
+        
+        self.sharedRequestCredentials = sharedRequestCredentials
+        if let credential = sharedRequestCredentials, let authToken = credential.authorizationToken {
+            sharedRequestHeaders["Authorization"] = "\(credential.tokenType) \(authToken)"
+        }
+        
+        guard savingToKeychain else { return }
+        CredentialStore.store(credential: sharedRequestCredentials, identifier: authenticator?.authIdentifier ?? "thundertable.com.threesidedcube-\(sharedBaseURL)", accessibility: accessibility)
+    }
+}
+
+extension RequestController {
+    
+    fileprivate static func showApplicationActivityIndicator() {
+        #if os(iOS)
+        if let option = Bundle.main.object(forInfoDictionaryKey: "TSCThunderRequestShouldHideActivityIndicator") as? Bool, !option {
```

Comment:
Just double checking this, is this definitely the right key? Also, would be good to move this to a constant :)